framework/contracts/native/account-factory/tests/create_account.rs

```diff
@@ -325,6 +332,76 @@ fn create_one_account_with_namespace() -> AResult {
         Some(String::from("account_description")),
         Some(String::from("https://account_link_of_at_least_11_char")),
         Some(namespace_to_claim.to_string()),
+        &[],
+    )?;
+
+    let manager_addr = account_creation.event_attr_value(ABSTRACT_EVENT_TYPE, "manager_address")?;
+    let proxy_addr = account_creation.event_attr_value(ABSTRACT_EVENT_TYPE, "proxy_address")?;
+
+    // We need to check if the namespace is associated with this account
+    let namespace = version_control.namespace(Namespace::new(namespace_to_claim)?)?;
+
+    assert_that!(&namespace).is_equal_to(&NamespaceResponse {
+        account_id: TEST_ACCOUNT_ID,
+        account_base: AccountBase {
+            manager: Addr::unchecked(manager_addr),
+            proxy: Addr::unchecked(proxy_addr),
+        },
+    });
+
+    Ok(())
+}
+
+#[test]
+fn create_one_account_with_namespace_fee() -> AResult {
+    let sender = Addr::unchecked(common::OWNER);
+    let chain = Mock::new(&sender);
+    let deployment = Abstract::deploy_on(chain.clone(), sender.to_string())?;
+
+    let factory = &deployment.account_factory;
+    let version_control = &deployment.version_control;
+
+    // Update namespace fee
+    let namespace_fee = coin(10, "token");
+    chain.set_balance(&sender, vec![namespace_fee.clone()])?;
+    version_control.update_config(None, Some(namespace_fee.clone()))?;
+
+    let namespace_to_claim = "namespace-to-claim";
+
+    let err = factory.create_account(
+        GovernanceDetails::Monarchy {
+            monarch: sender.to_string(),
+        },
+        vec![],
+        String::from("first_account"),
+        None,
+        None,
+        Some(String::from("account_description")),
+        Some(String::from("https://account_link_of_at_least_11_char")),
+        Some(namespace_to_claim.to_string()),
+        // Account creation fee not covered
+        &[],
+    );
+    assert!(err
+        .unwrap_err()
+        // Error type is inside contract, not the package
+        .root()
+        .to_string()
+        .contains("Invalid fee payment sent."));
```

Comment:
@CyberHoward wdyt do we want to move error types to packages as well?

Reply:
Error types of the contract? I'd prefer them to stay in the contract.